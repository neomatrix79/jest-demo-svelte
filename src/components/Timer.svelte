<script lang="ts">
  import { onMount } from "svelte";

  export let propsString: string;

  let timerString: string;

  onMount(() => {
    const timerId = setInterval(() => {
      timerString = new Date().toLocaleTimeString();
    }, 1000);

    return () => clearInterval(timerId);
  });
</script>

<div class="timer">
  {timerString}
  <p>{propsString}</p>
</div>
<button>ok</button>
<button data-testid="cancel">cancel</button>
<p>paragraph</p>

<style>
  .timer {
    font-size: 3.5rem;
    font-weight: 700;
    color: indigo;
    text-align: center;
    margin-top: 5rem;
  }
</style>
