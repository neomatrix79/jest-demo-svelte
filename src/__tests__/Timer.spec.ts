/**
 * @jest-environment jsdom
 */
// import { render } from "@testing-library/svelte";
// import Timer from "../components/Timer.svelte";

// test("render local timer", () => {
//   const { getByText, container, getByTestId } = render(Timer, {
//     propsString: "Timer",
//   });
//   const btn = document.querySelector("button");
//   const para = container.querySelector("p");
//   const cancel = getByTestId("cancel");

//   expect(getByText("Timer"));
//   expect(btn).not.toBeNull();
//   expect(cancel);
//   expect(para).not.toBeNull();
// });

import { render } from "@testing-library/svelte";
import Timer from "../components/Timer.svelte";

test("should match snapshot", () => {
  const { container } = render(Timer);

  expect(container).toMatchSnapshot();
});
