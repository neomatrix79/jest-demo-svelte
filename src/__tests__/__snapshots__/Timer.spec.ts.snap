// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<body>
  <div>
    <div
      class="timer svelte-wrx3ph"
    >
      undefined
       =&gt; 
      undefined
    </div>
  </div>
</body>
`;
